src/BackgroundLocationTracking/MainPage.xaml.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Esri.ArcGISRuntime.Location;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Controls;
+using System;
+using System.Threading.Tasks;
+using Location = Esri.ArcGISRuntime.Location.Location;
+
+namespace BackgroundLocationTracking
+{
+    public partial class MainPage : ContentPage
+    {
+        private LocationDataSource? _locationDataSource;
+
+        public MainPage()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            try
+            {
+#if IOS
+                _locationDataSource = new SystemLocationDataSource
+                {
+                    AllowsBackgroundLocationUpdates = true
+                };
+#else
+                _locationDataSource = new SystemLocationDataSource();
+#endif
+            }
+            catch (Exception ex)
+            {
+                // Handle initialization errors
+                LogMessage.Text = $"Error: An error occurred during initialization: {ex.Message}";
+            }
+        }
+
+        /// <summary>
+        /// Starts the location service.
+        /// </summary>
+        private async void StartTracking(object sender, EventArgs e)
+        {
+            if (_locationDataSource is null ||
+                await CheckAndRequestLocationPermission() is not PermissionStatus.Granted)
+            {
+                return; // Exit if the location data source is not initialized.
```

Comment:
nit: Not just if not initialized, but also if access was granted. Also I don't see how the location datasource could not be initialized so this null check doesn't make much sense.